index.js

```diff
@@ -64,9 +71,13 @@ export function persistentAtom(name, initial = undefined, opts = {}) {
   }
 
   onMount(store, () => {
-    store.set(storageEngine[name] ? decode(storageEngine[name]) : initial)
+    let sync = () => {
+      store.set(storageEngine[name] ? decode(storageEngine[name]) : initial)
+    }
+
+    sync()
     if (opts.listen !== false) {
-      eventsEngine.addEventListener(name, listener)
+      eventsEngine.addEventListener(name, listener, sync)
```

Comment:
Should we change types also? https://github.com/nanostores/persistent/blob/main/index.d.ts#L15

Reply:
I am not sure about that. We only need it in `windowPersistentEvents`. As far as I understand you can use nanostores in other environment (non browsers). So, it would not be needed there. We can probably type it that it is a "readonly" (you can't overwrite it anyway, we are setting it inside `onMount`) argument if you are working in browser.

Reply:
Yes, most users will not use it, but we need to specify it.

Reply:
Done